### PR TITLE
[5.3] Deprecate AbstractView::get()

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
In a time far far away, someone thought it might be a good idea to introduce an indirection in the view to get the data from the model by adding `AbstractView::get()`, which retrieves the model and then runs `get<Parameter>()` on the model. This removes any chance of typehinting, etc. and makes everything unnecessarily complex, so we are deprecating this method with this PR.

The new way hence forth shall be for example:
```
    public function display($tpl = null)
    {
        $model = $this->getModel();
        $this->items = $model->getItems();

        parent::display($tpl);
    }
```

### Testing Instructions
Codereview

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [X] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/318
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Review by laoneo — APPROVED** [review]

**Comment by laoneo** [comment]
Can you make here the manual entry as well?

**Comment by Hackwar** [comment]
The PR for manual.joomla.org has been created and linked. The question is how to get all the occurences of this in docs.joomla.org removed? Since there are plans on changing docs.joomla.org, should we change it there now?

**Comment by Hackwar** [comment]
I added a deprecation trigger to the method.

File: libraries/src/MVC/View/AbstractView.php [hunk 1]
```diff
@@ -138,9 +138,21 @@ abstract public function display($tpl = null);
...
+     * @deprecated __DEPLOY_VERSION__ will be removed in 8.0.
+     *              Retrieve the model with $model = $this->getModel(); and call the methods
+     *              to the model directly, e.g. $model->getItems() instead of $this->get('Items').
      */
     public function get($property, $default = null)
     {
+        trigger_deprecation(
+            'joomla/mvc/view',
+            '5.3',
+            'The %s() method is deprecated and will be removed in 7.0. use $model = $this->getModel();
```
Comment (Quy) [inline libraries/src/MVC/View/AbstractView.php:151]: 7.0 or 8.0?

File: libraries/src/MVC/View/AbstractView.php [hunk 2]
```diff
@@ -138,9 +138,21 @@ abstract public function display($tpl = null);
      * @return  mixed  The return value of the method
      *
      * @since   3.0
+     *
+     * @deprecated __DEPLOY_VERSION__ will be removed in 8.0.
```
Comment (laoneo) [inline libraries/src/MVC/View/AbstractView.php]:
```suggestion
     * @deprecated __DEPLOY_VERSION__ will be removed in 7.0.
```

File: libraries/src/MVC/View/AbstractView.php (hunk 1 above)
Comment (laoneo) [reply libraries/src/MVC/View/AbstractView.php:151]: 7 is ok

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**